aten/src/THC/generic/THCTensorCopy.cpp

```diff
@@ -58,6 +58,13 @@ void THTensor_(copyCuda)(THCState *state, THTensor *self, struct THCTensor *src)
 
   {
     THTensor *selfc = THTensor_(newContiguous)(self);
+    int tensorDevice = THCTensor_(getDevice)(state, src);
```

Comment:
Here would be a good place to put a note about how to do this correctly re current device.